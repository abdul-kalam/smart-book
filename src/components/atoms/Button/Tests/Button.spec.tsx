import React from "react";
import { shallow } from "enzyme";
import Button from "../Button.impl";

describe("Button component", () => {
    const clickFn = jest.fn();
    it("should render if Text props is not provided", () => {
        const component = shallow(<Button text={undefined} />);
        expect(component).toMatchSnapshot();
    });

    it("renders if Text props is provided", () => {
        const component = shallow(<Button text="hello" />);
        expect(component).toMatchSnapshot();
    });

    it("renders if Text props is filled with a span emojis", () => {
        const component = shallow(
            <Button
            text={
                    <span role="img" aria-label="so cool">
                        😀 😎 👍 💯
                    </span>
                }
            />,
        );
        expect(component).toMatchSnapshot();
    });

    it("should render if Text props is filled with emojis", () => {
        const component = shallow(<Button text="😀 😎 👍 💯" />);
        expect(component).toMatchSnapshot();
    });

    it("should be clickable", () => {
        const component = shallow(<Button onClick={clickFn} text="Login" />);

        component.find("Styled(Button)").simulate("click");

        expect(clickFn).toHaveBeenCalled();
    });
});
