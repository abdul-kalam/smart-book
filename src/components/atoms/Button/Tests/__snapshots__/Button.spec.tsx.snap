// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders if Text props is filled with a span emojis 1`] = `
<styled.button>
  <styled.span>
    <span
      aria-label="so cool"
      role="img"
    >
      😀 😎 👍 💯
    </span>
  </styled.span>
</styled.button>
`;

exports[`Button component renders if Text props is provided 1`] = `
<styled.button>
  <styled.span>
    hello
  </styled.span>
</styled.button>
`;

exports[`Button component should render if Text props is filled with emojis 1`] = `
<styled.button>
  <styled.span>
    😀 😎 👍 💯
  </styled.span>
</styled.button>
`;

exports[`Button component should render if Text props is not provided 1`] = `
<styled.button>
  <styled.span />
</styled.button>
`;
